_adr/80/index.adoc

```diff
@@ -1,22 +1,89 @@
+organisation.  In order to assist the customers, members of {glossary-customersupport} require a mechanism that allows them to capture the state of the
+instance of the service for diagnostic purposes.
+
+{glossary-customersupport} will use the gathered information in conjunction with the problem statement provided by the customer in order to efficiently
+investigate the problem and determine a solution.
 
-The data gathering mechanism must consider the issue of Personally Identifiable Information (PII) or other sensitive
-information and ensure that it is handled appropriately.
+The data gathering mechanism must consider the issue of Personally Identifiable Information (PII) or other sensitive information and ensure
+that it is handled appropriately.
 
 Authorization controls must be in place to ensure that the diagnostic gathering mechanism can only be used by suitably
 permissioned groups.
 
+## Goals
+
+* Identify the preferred solution for the diagnostic gathering mechanism.
+* Provide guidelines for content that should be included.
+
+## Non-goals
+
+None
+
+## Stakeholders
+* {glossary-businessstakeholder}
+* {glossary-sre}
+* {glossary-customersupport}
+
+## Current Architecture
+
+None
+
+## Proposed Architecture
+
+https://datatracker.ietf.org/doc/html/rfc2119[RFC-2119] keywords are used in this proposal.
+
+It is RECOMMENDED that managed services use https://source.redhat.com/groups/public/openshiftplatformsre/wiki/backplane_scripts[Back Plane]
+https://github.com/openshift/managed-scripts/[managed scripts] as the foundation of the data gathering mechanism. Backplane has an
+established https://source.redhat.com/groups/public/openshiftplatformsre/wiki/backplane_cluster_permissions[authorization model] that ensures
+that members of {glossary-customersupport} have the permissions to run the scripts.
+
+Each Managed Service that implements this ADR MUST provide one or more backplane scripts that interrogate the state of a service instance.
+The exact way way the scripts will interrogate state will be service specific, however if the service provided on top of OpenShift, it is
+RECOMMENDED that the scripts invoke the `oc adm inspect` command to gather Kubernetes state in addition to any other service specific
+state gathering commands.
+
+It is RECOMMENDED that the output from the scripts be a UNIX tar stream.  This will allow the artifact to provide the diagnostic
+information in a structured form that can easily be processed using common command-line tools.
+
+It is RECOMMENDED that the scripts takes reasonable steps to avoid the collection of PII or other sensitive information. This may be done
+either by redacting artifacts known to contain secrets/keys etc, or not gathering certain information in the first place.  For
+example, in a Java based system where PII may be present in the Java heap, the safe option is not to collect heap dumps.
```

Comment:
```suggestion
By default, it is RECOMMENDED that the scripts takes reasonable steps to avoid the collection of PII or other sensitive information. This may be done either by redacting artifacts known to contain secrets/keys etc, or not gathering certain information in the first place. The scripts MAY offer alternative modes of operations that collect deeper information.  The RECOMMENDED mechanism to enable these modes is to make use of [managed script parameters](https://github.com/openshift/managed-scripts/blob/main/scripts/SREP/example/metadata.yaml#L30).  If the use of such a mode may result in a artifacts containing PII or other sensitive information, the parameter's description MUST document that fact.
```